### ISSUE TITLE
Intellisense broken in VisualFSharp solution in VS

### ISSUE THREAD

**Issue by 0101** [dotnet/fsharp#17794]
Please provide a succinct description of the issue.

**Repro steps**

Provide the steps required to reproduce the problem:

1. Open VisualFSharp solution in VS
2. Go to any file in FSharp.Editor project (always seems to happen there but usually also other projects)

**Expected behavior**

We semantic coloring, type hints, tooltips, F12 works, etc.

**Actual behavior**

None of that works

**Known workarounds**

Edit all .fsproj files (e.g. add a whitespace character anywhere). 

**Related information**

VS 17.12 p2, but likely also other versions.

**Comment by 0101** [comment]
This seems to be happening since we started using SDK9 preview 7 in the repo.

The cause is that we don't get command line arguments from CPS for a few of the loaded projects. Therefore we cannot create FSharp options for them, or any that depend on them.

Digging further, CPS doesn't send us the command line args, because they are not populated into `FscCommandLineArgs` variable during design time build. 

This happens in `CoreCompile` target.

![image](https://github.com/user-attachments/assets/4d96656e-3b49-4875-ac12-b7fd902902df)

Here we see that for `FSharp.Core netstandard2.0` this target was skipped. Reason being outputs are up to date. 

These are the outputs
```
        Outputs="@(DocFileItem);
                 @(IntermediateAssembly);
                 @(IntermediateRefAssembly);
                 @(_DebugSymbolsIntermediatePath);
                 $(NonExistentFile);
                 @(CustomAdditionalCompileOutputs)"
```

One of them is `NonExistentFile` which depends on [`_ComputeNonExistentFileProperty`](https://github.com/dotnet/msbuild/blob/87d662a198e74b2d9e7f503e6ca21231900b1031/src/Tasks/Microsoft.Common.CurrentVersion.targets#L3855-L3883)

And we can see that one is skipped for the projects where we get the issue:

```
 Target "_ComputeNonExistentFileProperty" skipped, due to false condition; (('$(BuildingInsideVisualStudio)' == 'true') and ('$(BuildingOutOfProcess)' != 'true') and (('$(BuildingProject)' == 'false') or ('$(UseHostCompilerIfAvailable)' == 'true'))) was evaluated as (('true' == 'true') and ('' != 'true') and (('true' == 'false') or ('' == 'true'))).

Condition Analyzer:
❌ '$(BuildingProject)'=='false' ➔ 'true'=='false'
❌ '$(UseHostCompilerIfAvailable)'=='true' ➔ ''=='true'
```

`BuildingProject` should be `false` but it's `true`. And it's set via an unrelated project:

![image](https://github.com/user-attachments/assets/e68ba7dc-603c-4351-92cc-6b7991b65ac9)

**Comment by xperiandri** [comment]
@T-Gro, is this planned as a separate work, or will some other large feature fix this automatically when completed?

**Comment by T-Gro** [comment]
The biggest work item we have for this year is replacement of FSharp.Editor (this is what is added to VisualFsharp.sln when compared with the small solutions like FSharp.sln).

You are correct = this will both change composition of projects in our solution, as well as how F# in general (for any solution) behaves in VS.

**Comment by majocha** [comment]
Looks like this fixed itself and is no longer an issue.

**Comment by xperiandri** [comment]
Not fixed for me with `-windows10.x.x.x` targets

**Comment by majocha** [comment]
> Not fixed for me with `-windows10.x.x.x` targets

Yes, I can finally see this now with Elmish Samples.sln. The cause is different than this here issue, though.

**Comment by T-Gro** [comment]
@0101 has a script for mere touching of all .fsproj files, which re-kicked Intellisense (reloaded FSharpChecker instances).
I think it is a different issue :(

**Comment by majocha** [comment]
Yep, see https://github.com/dotnet/fsharp/issues/17440#issuecomment-3106469807
